### DISCUSSION TITLE
Expander Header Height

### DISCUSSION THREAD

**Discussion by xperia-droid** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3052] [Q&A] [answered]
Hi, 
i just want to adjust the Expander Header Height Value, so that's it smaller, as ~ 40 pixel.

I added the materialDesign Expander in my app.xaml and the MaterialDesignHorizontalHeaderStyle into it, but I don't found any way to make the Header smaller.

I tried different properties, but no change at all.

`<Style x:Key="CustomHeader" TargetType="{x:Type ToggleButton}">
                <Setter Property="Background" Value="Transparent" />

                <Setter Property="Template">
                    <Setter.Value>
                        <ControlTemplate TargetType="{x:Type ToggleButton}">
                            <Border Padding="{Binding Path=(wpf:ExpanderAssist.HorizontalHeaderPadding), RelativeSource={RelativeSource AncestorType=Expander, AncestorLevel=1}}" Background="{TemplateBinding Background}">
                                <Grid>
                                    <Grid.ColumnDefinitions>
                                        <ColumnDefinition Width="*" />
                                        <ColumnDefinition Width="16" />
                                        <ColumnDefinition Width="Auto" />
                                    </Grid.ColumnDefinitions>

                                    <ContentPresenter Grid.Column="0"
                                VerticalAlignment="Center"
                                Content="{TemplateBinding Content}"
                                ContentStringFormat="{TemplateBinding ContentStringFormat}"
                                ContentTemplate="{TemplateBinding ContentTemplate}"
                                ContentTemplateSelector="{TemplateBinding ContentTemplateSelector}" />

                                    <ToggleButton x:Name="ExpanderButton"
                            Grid.Column="2"
                            VerticalAlignment="Center"
                            Foreground="{TemplateBinding Foreground}"
                            IsChecked="{Binding Path=IsChecked, Mode=TwoWay, RelativeSource={RelativeSource TemplatedParent}}"
                            RenderTransformOrigin="0.5 0.5">
                                        <ToggleButton.Style>
                                            <Style TargetType="{x:Type ToggleButton}" BasedOn="{StaticResource MaterialDesignExpanderToggleButton}">
                                                <Style.Triggers>
                                                    <DataTrigger Binding="{Binding ExpandDirection, RelativeSource={RelativeSource AncestorType=Expander, AncestorLevel=1}}" Value="Up">
                                                        <Setter Property="RenderTransform">
                                                            <Setter.Value>
                                                                <RotateTransform Angle="180" />
                                                            </Setter.Value>
                                                        </Setter>
                                                    </DataTrigger>
                                                </Style.Triggers>
                                            </Style>
                                        </ToggleButton.Style>
                                    </ToggleButton>
                                </Grid>
                            </Border>
                        </ControlTemplate>
                    </Setter.Value>
                </Setter>
            </Style>

            <Style TargetType="{x:Type Expander}" x:Key="CustomExpanderStyle" BasedOn="{StaticResource MaterialDesignExpander}">
                <Setter Property="Template">
                    <Setter.Value>
                        <ControlTemplate TargetType="{x:Type Expander}">
                            <Border x:Name="Bd" BorderThickness="{TemplateBinding BorderThickness}" Background="{TemplateBinding Background}">
                                <Grid>
                                    <Grid.RowDefinitions>
                                        <RowDefinition Height="50" />
                                        <RowDefinition Height="*" />
                                    </Grid.RowDefinitions>
                                    <ToggleButton x:Name="HeaderSite" Grid.Row="0" Content="{TemplateBinding Header}" IsChecked="{Binding IsExpanded, Mode=TwoWay, RelativeSource={RelativeSource TemplatedParent}}" Style="{StaticResource CustomHeader}"/>
                                    <ContentPresenter x:Name="ExpandSite" Grid.Row="1" Visibility="Collapsed" Content="{TemplateBinding Content}" ContentTemplate="{TemplateBinding ContentTemplate}"/>
                                </Grid>
                            </Border>
                        </ControlTemplate>
                    </Setter.Value>
                </Setter>
            </Style>`

Thank you very much for your help :).

**Comment by nicolaihenriksen** [comment, marked as answer]
@xperia-droid Looking at the code you provided, one problem could be that you have given your `Expander` style and explicit key (i.e. "CustomExpanderStyle" string) rather than an implicit key (`Expander` type). When doing that, you will need to manually apply that custom style on every `Expander` instance by setting the `Style` property with something akin to `{StaticResource CustomExpanderStyle}`.

To avoid having to do that, you should be able to rely on the implicit style, so basically this part:
```xaml
<Style TargetType="{x:Type Expander}" x:Key="CustomExpanderStyle" BasedOn="{StaticResource MaterialDesignExpander}">
```
Should be (i.e. without the `x:key="CustomExpanderStyle"`):
```xaml
<Style TargetType="{x:Type Expander}" BasedOn="{StaticResource MaterialDesignExpander}">
```
